langchain/src/vectorstores/prisma.ts

```diff
@@ -0,0 +1,249 @@
+    function create<
+      TPrisma extends PrismaNamespace,
+      TColumns extends ModelColumns<TModel>
+    >(
+      config: {
+        prisma: TPrisma;
+        tableName: keyof TPrisma["ModelName"] & string;
+        vectorColumnName: string;
+        columns: TColumns;
+      },
+      embeddings: Embeddings
+    ) {
+      type ModelName = keyof TPrisma["ModelName"] & string;
+      return new PrismaVectorStore<TModel, ModelName, TColumns>(
+        { db, ...config },
+        embeddings
+      );
+    }
+
+    return { create };
+  }
+
+  async addModels(models: TModel[]) {
+    return this.addDocuments(
+      models.map((metadata) => {
+        const pageContent = typeof metadata[this.contentColumn];
+        if (pageContent !== "string")
+          throw new Error("Content column must be a string");
+        return new Document({ pageContent, metadata });
+      })
+    );
+  }
+
+  async addDocuments(documents: Document<TModel>[]) {
+    const texts = documents.map(({ pageContent }) => pageContent);
+    return this.addVectors(
+      await this.embeddings.embedDocuments(texts),
+      documents
+    );
+  }
+
+  async addVectors(vectors: number[][], documents: Document<TModel>[]) {
+    const idSql = this.Prisma.raw(`"${this.idColumn}"`);
+
+    await this.db.$transaction(
+      vectors.map(
+        (vector, idx) => this.db.$executeRaw`
+          UPDATE ${this.tableSql}
+          SET ${this.vectorColumnSql} = ${`[${vector.join(",")}]`}::vector
+          WHERE ${idSql} = ${documents[idx].metadata[this.idColumn]}
+        `
+      )
+    );
+  }
+
+  async similaritySearch(
+    query: string,
+    k = 4
+  ): Promise<Document<SimilarityModel<TModel, TSelectModel>>[]> {
+    const results = await this.similaritySearchVectorWithScore(
+      await this.embeddings.embedQuery(query),
+      k
+    );
+
+    return results.map((result) => result[0]);
+  }
+
+  async similaritySearchVectorWithScore(
+    query: number[],
+    k: number
+  ): Promise<[Document<SimilarityModel<TModel, TSelectModel>>, number][]> {
+    const vectorQuery = `[${query.join(",")}]`;
+    const articles = await this.db.$queryRaw<
+      Array<SimilarityModel<TModel, TSelectModel>>
+    >`
+      SELECT ${this.selectSql}, ${this.vectorColumnSql} <=> ${vectorQuery}::vector as "_distance" 
+      FROM ${this.tableSql}
+      ORDER BY "_distance" ASC
+      LIMIT ${k};
+    `;
+
+    const results: [Document<SimilarityModel<TModel, TSelectModel>>, number][] =
+      [];
+    for (const article of articles) {
+      if (article._distance != null) {
+        results.push([
+          new Document({
+            pageContent: article[this.contentColumn] as string | undefined,
+            metadata: article,
+          }),
+          article._distance,
+        ]);
+      }
+    }
+
+    return results;
+  }
+
+  /**
+   * @deprecated Not implemented in Prisma vectorstore, please use PrismaVectorStore.withModel instead
+   */
+  static fromTexts(
```

Comment:
These are the main methods we see people using, worth implementing I think. Given you implemented `addDocuments` doing these should be trivial, I'm happy to do them too